packages/block-editor/src/hooks/utils.js

```diff
@@ -19,3 +29,56 @@ export const cleanEmptyObject = ( object ) => {
 	);
 	return isEmpty( cleanedNestedObjects ) ? undefined : cleanedNestedObjects;
 };
+
+export function immutableSet( object, path, value ) {
+	return setWith( object ? clone( object ) : {}, path, value, clone );
+}
+
+export function transformStyles(
+	activeSupports,
+	migrationPaths,
+	result,
+	source,
+	index,
+	results
+) {
+	// If the condition verifies we are probably in the presence of a wrapping transform
+	// e.g: nesting paragraphs in a group or columns and in that case the styles should not be transformed.
+	if ( results.length === 1 && result.innerBlocks.length === source.length ) {
+		return result;
+	}
+	// For cases where we have a transform from one block to multiple blocks
+	// or multiple blocks to one block we apply the styles of the first source block
+	// to the result(s).
+	let referenceBlockAttributes = source[ 0 ]?.attributes;
+	// If we are in presence of transform between more than one block in the source
+	// that has more than one block in the result
+	// we apply the styles on source N to the result N,
+	// if source N does not exists we do nothing.
+	if ( results.length > 1 && source.length > 1 ) {
+		if ( source[ index ] ) {
+			referenceBlockAttributes = source[ index ]?.attributes;
+		} else {
+			return result;
```

Comment:
Can you provide an example that can reach to this code? I couldn't think of one to test.

Reply:
We don't have any core transform where this happens but it may happen on a third-party block so I'm guarding against it. This covers a case where multiple blocks are transformed into an even higher number of blocks.